apps/web/src/app/(signing)/sign/[token]/form.tsx

```diff
@@ -121,16 +121,17 @@ export const SigningForm = ({ document, recipient, fields }: SigningFormProps) =
             </div>
 
             <div className="flex flex-col gap-4 md:flex-row">
-              <Button
-                type="button"
-                className="dark:bg-muted dark:hover:bg-muted/80 w-full  bg-black/5 hover:bg-black/10"
-                variant="secondary"
-                size="lg"
-                disabled={typeof window !== 'undefined' && window.history.length <= 1}
-                onClick={() => router.back()}
-              >
-                Cancel
-              </Button>
+              {window.history.length > 1 && (
```

Comment:
`typeof window` needs to be checked to avoid SSR issues, realistically this shouldn't be any different than the disable statement above so it should still fail in the same situations the issue reported.

Reply:
I guess it will always be disabled in Gmail because Gmail always opens the document in a new tab and it has no history. 
I removed `typeof window` because in dev server, it was showing error "Error: Text content does not match server-rendered HTML.Warning: Text content did not match. Server: "Complete" Client: "Cancel"
"

Reply:
That would be the intention, it should be disabled for people opening from the email while those accessing it via the dashboard (inbox) would have the ability to cancel and go back.

Again I still think disabling is the way to go here, otherwise to handle this in an SSR friendly manner you'll need to use state and an effect to deal with hydration mismatches.

Reply:
Okay, I'll leave it as it is. I initially thought I needed to resolve this issue since it was assigned to me. I'll close this pull request and the associated issue.